src/Build/Evaluation/ItemSpec.cs

```diff
@@ -423,14 +423,12 @@ private FileSpecMatcherTester FileMatcher
         {
             get
             {
-                if (_fileMatcherInitialized)
+                if (!_fileMatcherInitialized)
                 {
-                    return _fileMatcher;
+                    _fileMatcher = CreateFileSpecMatcher();
+                    _fileMatcherInitialized = true;
                 }
 
-                _fileMatcher = CreateFileSpecMatcher();
-                _fileMatcherInitialized = true;
-
                 return _fileMatcher;
```

Comment:
What are the relative advantages/disadvantages to doing so? And it would just be replacing `return _fileMatcher;` with `return ref _fileMatcher` and `private FileSpecMatcherTester FileMatcher` with `private ref FileSpecMatcherTester FileMatcher`?

Reply:
```C#
    struct MyStruct
    {
        public int Field1;
        public int Field2;
        public int Field3;
        public int Field4;
    }

    class MyClass
    {
        MyStruct _struct;

        public ref MyStruct Struct
        {
            get
            {
                return ref _struct;
            }
        }
    }
```

The `Struct` getter in this example runs about 4x faster compared to a version without the `ref` because it's not copying the struct. Since in your case you're not mutating the returned struct (it can even be declared `readonly`), there wouldn't be any downside.

> And it would just be replacing `return _fileMatcher;` with `return ref _fileMatcher` and `private FileSpecMatcherTester FileMatcher` with `private ref FileSpecMatcherTester FileMatcher`?

Yes to the former, no to the latter. The `ref` should be added to the `return` statement and to the property.